### DISCUSSION TITLE
How can I use horizontal scrolling in neovim?

### DISCUSSION THREAD

**Discussion by TylerBarnes** [wezterm/wezterm#3749] [Q&A] [answered]
@tlinford I saw you made a PR (https://github.com/wez/wezterm/pull/3001) to add support for horizontal scroll events which is pretty awesome.
I'm having some issues figuring out how to use it though, any chance you might be able to give me a hand? 
I have line wrapping turned off in neovim so text spills off the right side of the screen.
I'm switching from kitty to wezterm and with nvim wrapping turned off in kitty horizontal scrolling works, however in wezterm it still doesn't work on the latest version.
I've been trying to figure it out but I think I don't know enough about how all this stuff works to get it.
Any ideas/tips?
Do I need to enable something for this to work?

**Comment by TylerBarnes** [comment]
Hmm not having much luck figuring this out 😅 any help is appreciated!

If I add this to my neovim config:
```
vim.api.nvim_set_keymap('n', '<ScrollWheelLeft>', ':echo "Scroll left event detected"<CR>', {})
vim.api.nvim_set_keymap('n', '<ScrollWheelRight>', ':echo "Scroll right event detected"<CR>', {})
```

In kitty it echo's the text on horizontal scroll, but nothing shows up with the same config in Wezterm.

**Reply by jdhao** [reply]
I tested this on latest wezterm release with nvim 0.9.1. It works for me. Also in neovim, there is [mousescroll](https://github.com/neovim/neovim/commit/93c8fe77cbfd3934c7599196e573136f75f6e9af) option that you might be interested.

**Reply by TylerBarnes** [reply]
Thank you, that's really helpful!

**Comment by wez** [comment, marked as answer]
The linked PR was to support the ability for termwiz applications to parse horizontal scroll events.
wezterm the terminal emulator doesn't currently encode horizontal mouse events to send them to the terminal.
Please open an enhancement issue for this!
